### PR TITLE
Fix server not shutting down when exiting early.

### LINKED ISSUE #6422: Server doesn't stop properly if EULA is false.
### Expected behavior

When the EULA isn't accepted, the server should print a message about it and shut down.

### Observed/Actual behavior

When the EULA isn't accepted, the server prints a message and stays running.

### Steps/models to reproduce

- Download paper 1.17.1, build 195. 
- Start the server without accepting the EULA.

### Plugin list

None

### Paper version

[15:43:24 INFO]: Checking version, please wait...
[15:43:25 INFO]: This server is running Paper version git-Paper-195 (MC: 1.17.1) (Implementing API version 1.17.1-R0.1-SNAPSHOT) (Git: fa184f7)
You are running the latest version

### Agreements

- [X] I am running the latest version of Paper available from https://papermc.io/downloads.
- [X] I have searched for and ensured there isn't already an open issue regarding this.
- [X] My version of Minecraft is supported by Paper.

### Other

Java Version:
openjdk version "16.0.1" 2021-04-20
OpenJDK Runtime Environment AdoptOpenJDK-16.0.1+9 (build 16.0.1+9)
OpenJDK 64-Bit Server VM AdoptOpenJDK-16.0.1+9 (build 16.0.1+9, mixed mode, sharing)

JStack: https://paste.gg/p/anonymous/3d49e1fbeefd420a91cd2be122e004e2

### PR DESCRIPTION
**Pull request by NoahvdAa**
Fixes #6422 with a workaround until a Log4J update is released.

### CODE REVIEW

File: patches/server/0224-Use-AsyncAppender-to-keep-logging-IO-off-main-thread.patch [hunk 1]
```diff
@@ -16,6 +16,46 @@ index 34a0d6b54a15c8aa0c706541316c5d448e3d94b9..76a8db5ff623b0a3b83a0861f642da67
...
+diff --git a/src/main/java/net/minecraft/server/Main.java b/src/main/java/net/minecraft/server/Main.java
+index fb0b3c5770f66cc3590f5ac4e690a33cb6179be3..1537e1b54564295acf0c0466156b4a1f63919922 100644
+--- a/src/main/java/net/minecraft/server/Main.java
++++ b/src/main/java/net/minecraft/server/Main.java
+@@ -112,6 +112,8 @@ public class Main {
+ 
+             if (optionset.has("initSettings")) { // CraftBukkit
+                 Main.LOGGER.info("Initialized '{}' and '{}'", path.toAbsolutePath(), path1.toAbsolutePath());
++                LogManager.shutdown(); // Paper - workaround until Log4J is updated.
++                Process.exit(0); // Paper - workaround until Log4J is updated.
```
Comment (lynxplay) [inline patches/server/0224-Use-AsyncAppender-to-keep-logging-IO-off-main-thread.patch]: Maybe my brain is failing me, but what method is `Process#exit`? This does not compile for me locally and I haven't heard of that method either 😅 

Was this supposed to be `System#exit` ?

Comment (NoahvdAa) [reply patches/server/0224-Use-AsyncAppender-to-keep-logging-IO-off-main-thread.patch]: Whoops, that's very wrong 😂 . I have no idea how I managed to miss that.

Comment (NoahvdAa) [reply patches/server/0224-Use-AsyncAppender-to-keep-logging-IO-off-main-thread.patch]: Fixed now!

**Review by lynxplay — CHANGES_REQUESTED** [review]
Concerning that the entire `main` method of the `Main` class is wrapped in a `try-catch` and the `catch` block also uses a logger, the workaround would have to be present there as well. (Check the bottom of the `main` method)

**Comment by NoahvdAa** [comment]
Closing in favor of #6424 .

**Pull request closed without merging**